pkg/controller/jobs/job/job_webhook.go

```diff
@@ -101,18 +83,15 @@ var _ webhook.CustomValidator = &JobWebhook{}
 // ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
 func (w *JobWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
 	job := obj.(*batchv1.Job)
-	return validateCreate(job).ToAggregate()
+	log := ctrl.LoggerFrom(ctx).WithName("job-webhook")
+	log.V(5).Info("Validating create", "job", klog.KObj(job))
+	return validateCreate(&Job{*job}).ToAggregate()
 }
 
-func validateCreate(job *batchv1.Job) field.ErrorList {
+func validateCreate(job jobframework.GenericJob) field.ErrorList {
 	var allErrs field.ErrorList
-	for _, crdNameAnnotation := range []string{constants.ParentWorkloadAnnotation, constants.QueueAnnotation} {
-		if value, exists := job.Annotations[crdNameAnnotation]; exists {
-			if errs := validation.IsDNS1123Subdomain(value); len(errs) > 0 {
-				allErrs = append(allErrs, field.Invalid(annotationsPath.Key(crdNameAnnotation), value, strings.Join(errs, ",")))
-			}
-		}
-	}
+	allErrs = append(allErrs, jobframework.ValidateCreateForCRDNameAnnotation(job, constants.ParentWorkloadAnnotation)...)
```

Comment:
maybe these constants make more sense as part of the framework now.

Reply:
Agree, move the ones directly used by the framework under the jobframework dir

Reply:
Nah, I have to revert it for now as I was getting cyclic imports.

Reply:
oh what else needs the constants?

Reply:
ok, I have looked into this. So once I move the constants this is one of two loops:
- pkg/util/priority/priority_test.go                      imports pkg/util/testing
- pkg/util/testing/wrappers.go                           imports pkg/controller/jobframework [new edge for constants]
- pkg/controller/jobframework/reconciler.go  imports pkg/util/priority

Reply:
https://github.com/kubernetes-sigs/kueue/pull/640